### ISSUE TITLE
NCR machines have buggy emulation

### ISSUE THREAD

**Issue by TheSystemGuy1337** [86Box/86Box#4814]
### What happened?

I wanted to mess around with the NCR 3302 for the fun of it, but ran into some issues, some of them are odd. First, the machine always starts off with 384 KB of extended memory in the BIOS. Attempting to correct this will fail upon POST with `Memory Error Bit 0000 FFFF`. Second issue is inserting floppy images will still show a grey icon, the disk indicator light won't turn on. The system still reads floppies fine, you just can't see activity anymore. Same issues apply to all the other NCR machines.

### Configuration file

```ini
[General]
vid_renderer = qt_opengl3
host_cpu = AMD Ryzen 5 5600H with Radeon Graphics         
emu_build_num = 6145
uuid = aff161ca-2db2-580b-b472-70c032465cea
video_gl_shader = G:/fakelottes.glsl
vid_resize = 2
window_fixed_res = 1024x768
force_43 = 1
enable_overscan = 1
update_icons = 0
iconset = winbox
video_grayscale = 3

[Machine]
machine = 3302
cpu_family = 286
cpu_speed = 16000000
cpu_multi = 1
cpu_use_dynarec = 0
fpu_softfloat = 0
time_sync = disabled
fpu_type = 287
mem_size = 4096

[Video]
gfxcard = internal

[Input devices]
mouse_type = msserial

[Sound]
fm_driver = nuked
sndcard = sbprov1

[Network]
net_01_link = 0
net_02_link = 0
net_03_link = 0
net_04_link = 0
net_01_net_type = slirp

[Storage controllers]
hdc = internal
cassette_mode = load

[Floppy and CD-ROM drives]
fdd_01_type = 35_2hd
fdd_02_type = none
fdd_01_turbo = 1
fdd_01_fn = C:/Users/TheSystemGuy/Desktop/Microsoft MS-DOS 5.00 (3.5-720k)/Disk01.img

[Hard disks]
hdd_01_parameters = 17, 11, 754, 0, ide
hdd_01_fn = G:/cashregister.vhd
hdd_01_speed = ramdisk
hdd_01_vhd_blocksize = 4096
hdd_01_ide_channel = 0:1

[Sound Blaster Pro v1 #1]
base = 0220
irq = 7
dma = 1
opl = 1
receive_input = 1
```

### Operating system

Windows 11

### CPU

AMD Ryzen 5 5600H

### 86Box version

4.3 build 6145

### Build architecture

Windows - x64 (64-bit)

### Build type

- [X] New recompiler
- [ ] Debug build

### Download source

Official website (Jenkins, GitHub)

### Additional context

NCR makes cash registers and ATMs, so I wanted to emulate one of those machines. Or it could be another NCR who made it, I don't know.

**Comment by OBattler** [comment]
Did you specify a FDC in Settings?

**Comment by OBattler** [comment]
OK, I can reproduce the floppy icon problem but not the memory problem.

**Comment by OBattler** [comment]
OK, go to Media and make sure Update status bar icons is ticked.

**Comment by OBattler** [comment]
OK, the emulator wasn't updating the empty/full icons when updating status bar icons was disabled. This is now fixed. For the activity, however, re-enable updating status bar icons.

As for the memory problem, I can't reproduce it at all. Make sure you're setting extended memory in CMOS setup to total memory - 1024 like I did - in your case, you have 4096 kB total memory, so set extended memory in CMOS setup to 3072.

As such, I'm now closing this.
